eth/stagedsync/stage_headers.go

```diff
@@ -651,7 +672,7 @@ func handleInterrupt(interrupt engineapi.Interrupt, cfg HeadersCfg, tx kv.RwTx,
 		if interrupt == engineapi.Stopping {
 			cfg.hd.PayloadStatusCh <- engineapi.PayloadStatus{CriticalError: errors.New("server is stopping")}
 		}
-		if interrupt == engineapi.Synced {
+		if interrupt == engineapi.Synced && cfg.hd.HeadersCollector() != nil {
```

Comment:
When is `HeadersCollector` `nil?`

Reply:
it can happen when we download in-place and the we interrupt and use the other instance of saveHeaderPos there, that is the case.